x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/Hasher.java

```diff
@@ -738,7 +738,7 @@ public static List<String> getAvailableAlgoStoredHash() {
         return Arrays.stream(Hasher.values())
             .map(Hasher::name)
             .map(name -> name.toLowerCase(Locale.ROOT))
-            .filter(name -> (name.startsWith("pbkdf2") || name.startsWith("bcrypt")))
+            .filter(name -> (name.startsWith("pbkdf2") || name.startsWith("bcrypt") || "ssha256".equals(name)))
```

Comment:
This needs to be allowed only for API keys, not for passwords (api keys are random long secrets).